src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -1295,11 +1466,22 @@ otError RadioSpinel<InterfaceType, ProcessContextType>::Get(spinel_prop_key_t aK
 
     assert(mWaitingTid == 0);
 
-    mPropertyFormat = aFormat;
     va_start(mPropertyArgs, aFormat);
-    error = RequestV(true, SPINEL_CMD_PROP_VALUE_GET, aKey, nullptr, mPropertyArgs);
+    error = RequestWithPropertyFormatV(aFormat, SPINEL_CMD_PROP_VALUE_GET, aKey, nullptr, mPropertyArgs);
     va_end(mPropertyArgs);
-    mPropertyFormat = nullptr;
+
+#if OPENTHREAD_SPINEL_CONFIG_RCP_RESTORATION_MAX_COUNT > 0
+    if (mRcpFailed)
+    {
+        RecoverFromRcpFailure();
+
+        va_start(mPropertyArgs, aFormat);
+        error = RequestWithPropertyFormatV(aFormat, SPINEL_CMD_PROP_VALUE_GET, aKey, nullptr, mPropertyArgs);
+        va_end(mPropertyArgs);
+
+        VerifyOrDie(!mRcpFailed, OT_EXIT_FAILURE);
```

Comment:
What if OPENTHREAD_SPINEL_CONFIG_RCP_RESTORATION_MAX_COUNT >= 2 and RCP failed here?

We should restore from this RCP failure too since it's the 2nd fail.